bluecellulab/reports/writers/spikes.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, List
+from bluecellulab.reports.writers.base_writer import BaseReportWriter
+import logging
+import os
+import numpy as np
+import h5py
+
+logger = logging.getLogger(__name__)
+
+
+class SpikeReportWriter(BaseReportWriter):
+    """Writes SONATA spike report from pop→gid→times mapping."""
+
+    def write(self, spikes_by_pop: Dict[str, Dict[int, list]]):
+        if os.path.exists(self.output_path):
```

Comment:
since output_path is pathlib, you can use the pathlib functions instead of os.path and os.remove